### ISSUE TITLE
rqt_plot crashes when resizing the window

### ISSUE THREAD

**Issue by jonathan84clark** [ros-visualization/rqt_plot#54]
Steps to reproduce.
1. Type rqt_plot in a terminal window
2. Grab the bottom edge of the rqt_plot gui window with the mouse
3. Drag to resize.

I get this error:
`qurrent@qurrent-HP-Z820-Workstation:~/catkin_ws$ rqt_plot /vrx/debug/wind/speed
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/matplotlib/backends/backend_qt5agg.py", line 77, in paintEvent
    self.resizeEvent(event)
  File "/opt/ros/melodic/lib/python2.7/dist-packages/rqt_plot/data_plot/mat_data_plot.py", line 110, in resizeEvent
    self.figure.tight_layout()
  File "/usr/lib/python2.7/dist-packages/matplotlib/figure.py", line 2031, in tight_layout
    self.subplots_adjust(**kwargs)
  File "/usr/lib/python2.7/dist-packages/matplotlib/figure.py", line 1881, in subplots_adjust
    self.subplotpars.update(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/matplotlib/figure.py", line 241, in update
    raise ValueError('bottom cannot be >= top')
ValueError: bottom cannot be >= top`

OS: Ubuntu 18.04
Version: ros-melodic-rqt is already the newest version (0.5.1-1bionic.20200320.111301)

Any help here would be great!

Thanks,

Jonathan L Clark

**Comment by floesche** [comment]
Possibly related to this and #35, I see `rqt_plot` crashes for me on resize with python3.8, matplotlib-3.2.1, and ros2 eloquent. The error message I am seeing is:

```
Traceback (most recent call last):
  File "${ROS_WS}/install/rqt_plot/lib/python3.8/site-packages/rqt_plot/data_plot/mat_data_plot.py", line 109, in resizeEvent
    super(MatDataPlot.Canvas, self).resizeEvent(event)
  File "/usr/lib/python3.8/site-packages/matplotlib/backends/backend_qt5.py", line 397, in resizeEvent
    self.figure.set_size_inches(winch, hinch, forward=False)
  File "/usr/lib/python3.8/site-packages/matplotlib/figure.py", line 902, in set_size_inches
    raise ValueError(f'figure size must be positive finite not {size}')
ValueError: figure size must be positive finite not [9.33 0. ]
```

**Comment by robwaat** [comment]
I am also seeing the issue described by Johnathan with the following setup
- OS: Ubuntu:18.04, 
- Package: ros-melodic-rqt bionic,now 0.5.1-1bionic.20200320.111301 amd64 
- Backend: matplotlib

The error message was:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/matplotlib/backends/backend_qt5agg.py", line 77, in paintEvent
    self.resizeEvent(event)
  File "/opt/ros/melodic/lib/python2.7/dist-packages/rqt_plot/data_plot/mat_data_plot.py", line 110, in resizeEvent
    self.figure.tight_layout()
  File "/usr/lib/python2.7/dist-packages/matplotlib/figure.py", line 2031, in tight_layout
    self.subplots_adjust(**kwargs)
  File "/usr/lib/python2.7/dist-packages/matplotlib/figure.py", line 1881, in subplots_adjust
    self.subplotpars.update(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/matplotlib/figure.py", line 241, in update
    raise ValueError('bottom cannot be >= top')
ValueError: bottom cannot be >= top
```

I found that a workaround is to right-click on the rqt_plot window banner and maximize the window. Then right click again and un-maximize. Afterwards, the rqt_plot no longer seems to crash upon resizing. 

This "fix" appears to persist across multiple re-launches.

**Comment by ch3ll-n3m3s1s** [comment]
After installing pyqtgraph via pip, rqt_plot is no longer crashing on resize (and is correctly displaying a graph at startup). I am still using the matplotlib interface for presentation.

**Comment by Oemer-Terlemez** [comment]
>  After installing pyqtgraph via pip, rqt_plot is no longer crashing on resize (and is correctly displaying a graph at startup). I am still using the matplotlib interface for presentation.
Thank you very much. This fixed the issue for me also!

**Comment by sergiosierram** [comment]
> After installing pyqtgraph via pip, rqt_plot is no longer crashing on resize (and is correctly displaying a graph at startup). I am still using the matplotlib interface for presentation.

This also worked for me! I was running Ubuntu 18.04 and ROS melodic in a virtual machine in Oracle Virtual Box

**Comment by rhaschke** [comment]
This is a duplicate of #35, which was actually fixed in #52 nearly a year ago, becoming part of release >= 0.4.11.
However, this fix is still not released into Kinetic (stuck at 0.4.8) or Melodic (stuck at 0.4.9).
@dirk-thomas: Could you please prepare new releases, such that rqt_plot works out of the box with the system-installed matplotlib? Thanks.

**Comment by mabelzhang** [comment]
ROS 1 branch with the fix in #69 is pending release to kinetic, melodic, and noetic. Will be in the next respective syncs.
https://github.com/ros/rosdistro/pull/28396
https://github.com/ros/rosdistro/pull/28397
https://github.com/ros/rosdistro/pull/28398

I haven't looked at the ROS 2 fix yet but will try to soon.

**Comment by clalancette** [comment]
This fix was released to ROS 1 quite a while ago, and I also just retested this on the latest ROS 2 Rolling.  It seems to not occur there.  With all of that said, I'm going to close this out, but please feel free to reopen if this is still occurring for you.
